src/huggingface_hub/repocard.py

```diff
@@ -109,7 +109,9 @@ def content(self, content: str):
             data_dict = {}
             self.text = content
 
-        self.data = self.card_data_class(**data_dict, ignore_metadata_errors=self.ignore_metadata_errors)
+        self.data = self.card_data_class(
+            **data_dict, ignore_metadata_errors=self.ignore_metadata_errors, original_order=list(data_dict.keys())
+        )
```

Comment:
Hi @hlky, thanks for your PR! The logic of it looks good and seems to address at least part of the problem (metadata ordering). In a future PR we might want to address other styling issues (typically how to handle long strings). In that sense, I think it'd be best not to change the card data classes (e.g. `ModelCardData`) but instead keep the formatting constraint (e.g. the original order) in the `ModelCard` itself. 

What I suggest is:
```suggestion
        self.data = self.card_data_class(**data_dict, ignore_metadata_errors=self.ignore_metadata_errors)
        self._original_order = list(data_dict.keys())
```

and then in the `def content(self):` property, provide the `original_order` to `to_yaml`.

The result of this will be the same but logic-wise I prefer to have all the format constraints stored in the Card itself and the CardData is only a dataclass without hidden logic). What do you think? Would you be up to update the PR accordingly?

Reply:
Done, thanks.